src/map/battle.cpp

```diff
@@ -2868,12 +2868,12 @@ static bool is_skill_using_arrow(struct block_list *src, int32 skill_id)
 	if (sd == nullptr && skill_id != 0) {
 		if (skill_get_ammotype(skill_id) != 0)
 			return true;
+	}
 
-		status_data* sstatus = status_get_status_data(*src);
+	status_data* sstatus = status_get_status_data(*src);
 
-		if (sstatus->rhw.range > 3)
-			return true;
-	}
+	if (sstatus->rhw.range > 3)
```

Comment:
If I just look at the previous code. I think this should still be within the `sd == nullptr` area.

Breaking down the old formula:

1.
(sd && sd->state.arrow_atk) || 

2.
(!sd && (
	(skill_id && skill_get_ammotype(skill_id)) ||   //Skill ID is only checked for the skill_get_ammotype call
	sstatus->rhw.range>3
)
	
3.
|| skill_id == HT_FREEZINGTRAP || (skill_id == HT_PHANTASMIC) || (skill_id == GS_GROUNDDRIFT));